examples/key_map/main.ml

```diff
@@ -0,0 +1,48 @@
+open Leaves
+open Minttea
+
+type msg = CursorUp | CursorDown | CursorLeft
+
+let defaults =
+  let open Key_map in
+  make [
+    on
+      ~help:{ key = "up"; desc = "↑/k" }
+      [ Minttea.Event.Up; Minttea.Event.Key "k" ]
+      CursorUp;
+    on
+      ~help:{ key = "down"; desc = "↓/j" }
+      [ Minttea.Event.Down; Minttea.Event.Key "j" ]
+      CursorDown;
+    on ~disabled:true
+      ~help:{ key = "left"; desc = "←/h" }
+      [ Minttea.Event.Left; Minttea.Event.Key "h" ]
+      CursorLeft;
+  ]
+
+let custom_key_map =
+  let open Key_map in
+  [
+    on
+      ~help:{ key = "up"; desc = "↑/k" }
+      [ Minttea.Event.Up; Minttea.Event.Key "k"; Minttea.Event.Key "u" ]
+      CursorUp;
+  ]
+
+let () =
```

Comment:
Almost forgot, all the examples should be working Minttea app to showcase how this component fits into an application.

This also helps you see if the API really figs the way you'd use it in an app.

In this case we can also refactor an existing example that uses several keys (like the views example) so use one or more key maps.

This should also expose the component to the use cases of:

* how do you deal with multiple key maps in a single app?

* how do we render key maps for users to see that they are available?

We don't have to have those answers in this PR but they would definitely inform the design you go for.

Reply:
Ok, so in this case should I remove the key_map example and refactor say the views example to use the key map feature?

Reply:
Up to you :) either making this one a complete tiny app, or refactoring one of the others.